### ISSUE TITLE
Constraint Errors when using newConversationViewController

### ISSUE THREAD

**Issue by daresayer** [smooch/smooch-ios#66]
I get a few "Unable to simultaneously satisfy constraints" errors when using newConversationViewController. 

```
(
    "<NSLayoutConstraint:0x13c73a690 UIInputSetContainerView:0x13c6c77d0.bottom == UITextEffectsWindow:0x13c738dc0.bottom>",
    "<NSLayoutConstraint:0x13c6cbd20 V:[UIInputSetContainerView:0x13c6c77d0]-(0)-[UIInputSetHostView:0x13c6c8040]>",
    "<NSLayoutConstraint:0x13c6dff40 UITextEffectsWindow:0x13c738dc0.bottom >= SKTSOMessageInputView:0x13c6e76d0.bottom + 50>",
    "<NSLayoutConstraint:0x13db12ba0 SKTSOMessageInputView:0x13c6e76d0.bottom == UIInputSetHostView:0x13c6c8040.bottom>"
)
```

```
 Unable to simultaneously satisfy constraints.
	Probably at least one of the constraints in the following list is one you don't want. 
	Try this: 
		(1) look at each constraint and try to figure out which you don't expect; 
		(2) find the code that added the unwanted constraint or constraints and fix it. 
(
    "<NSLayoutConstraint:0x13c73a690 UIInputSetContainerView:0x13c6c77d0.bottom == UITextEffectsWindow:0x13c738dc0.bottom>",
    "<NSLayoutConstraint:0x13c698cd0 UITextEffectsWindow:0x13c738dc0.bottom >= SKTSOMessageInputView:0x13c6e76d0.bottom + 50>",
    "<NSLayoutConstraint:0x13c727e80 SKTSOMessageInputView:0x13c6e76d0.bottom == UIInputSetContainerView:0x13c6c77d0.bottom>"
)

Will attempt to recover by breaking constraint 
<NSLayoutConstraint:0x13c727e80 SKTSOMessageInputView:0x13c6e76d0.bottom == UIInputSetContainerView:0x13c6c77d0.bottom>

Make a symbolic breakpoint at UIViewAlertForUnsatisfiableConstraints to catch this in the debugger.
The methods in the UIConstraintBasedLayoutDebugging category on UIView listed in <UIKit/UIView.h> may also be helpful.

```

**Comment by wikyd** [comment]
I also get this error when using this method.  I'm trying to use this controller as a tab within a tab bar.  In my storyboard I have a custom UINavigation controller linked to the tab.  This custom class looks like this:

```swift
class SmoochWrapperViewController: UINavigationController {

  override func viewDidLoad() {
    super.viewDidLoad()
    let smoochController = Smooch.newConversationViewController()
    self.pushViewController(smoochController, animated: false)
  }
  
}
```

The UI appears fine, but I get a lot of errors in my logs.  Is there a flag I need to set when presenting this controller normally instead of modally?

**Comment by mspensieri** [comment]
@daresayer @wikyd Thanks for the report guys, finally getting a chance to look into this. The chat input is an `inputAccessoryView` of the conversation view controller, so it always wants to be docked at the bottom of the screen. The constraint that is conflicting is one we add to make sure the chat input is docked above the `UITabBar`, rather than covering it up.

I'll fiddle around with it and see if I can clear up the warnings.

btw @wikyd the `UINavigationController` wrapper shouldn't be necessary - unless you're using it for something else? You could just set the conversation VC as a child view controller of your tab controller directly

**Comment by mspensieri** [comment]
Should be fixed with v3.5.3

**Comment by wikyd** [comment]
Thanks!  This fixed the animation issue we talked about over email as well.
